automapper/mapper.py

```diff
@@ -205,16 +202,23 @@ def map(
 
         common_fields_mapping = fields_mapping
         if target_cls_field_mappings:
-            # transform mapping if it's from source class field
-            common_fields_mapping = {
-                target_obj_field: (
-                    getattr(obj, source_field[len(obj_type_prefix) :])
-                    if isinstance(source_field, str)
-                    and source_field.startswith(obj_type_prefix)
-                    else source_field
-                )
-                for target_obj_field, source_field in target_cls_field_mappings.items()
-            }
+            # Transform mapping if it's from source class field
+            common_fields_mapping = {}
+
+            for target_obj_field, source_field in target_cls_field_mappings.items():
+                if isinstance(source_field, str) and source_field.startswith(
+                    obj_type_prefix
+                ):
+                    common_fields_mapping[target_obj_field] = self._rgetter(
+                        obj, source_field[len(obj_type_prefix) :]
+                    )
+                elif isinstance(source_field, MapPath):
+                    common_fields_mapping[target_obj_field] = self._rgetter(
+                        obj, source_field
+                    )
```

Comment:
then here we know that we need to use recursive getter.
Could be also `resolve_map_path` or `get_by_map_path` or something similar, little bit more descriptive.
You'll know that you pass in a source_field of type MapPath and this function will get you a value using this MapPath.

If the value not possible to retrieve, I would suggest this function raises an `InvalidMapPathError`. So this is something user may expect from our package. Currently the reduce function in `_rgetter` would fail with unknown error.